src/checkers/inference/model/ComparisonConstraint.java

```diff
@@ -0,0 +1,133 @@
+package checkers.inference.model;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.javacutil.BugInCF;
+
+import com.sun.source.tree.Tree.Kind;
+
+/**
+ * Represents a constraint that two slots must be comparable.
```

Comment:
This is the description from `ComparableConstraint`. What is the difference between the two?
Please adapt this sentence and add an explanation of the difference to `ComparableConstraint`, to both classes.